src/cpp/session/modules/SessionThemes.cpp

```diff
@@ -362,6 +363,83 @@ Error getThemes(const json::JsonRpcRequest& request,
    return Success();
 }
 
+/**
+ * @brief RPC that lets the client add a theme for the current user.
+ *
+ * @param request       The request from the client to add a theme. The theme should already exist
+ *                      on the server and the only parameter on the request should be the location
+ *                      of the theme to add.
+ * @param pResponse     The response from the server. Will contain the name of the newly added theme.
+ *
+ * @return `Success` on success; an error otherwise.
+ */
+Error addTheme(const json::JsonRpcRequest& request,
+                     json::JsonRpcResponse* pResponse)
+{
+   std::string themeToAdd = request.params.at(0).get_str();
+
+   boost::regex rsThemeRegex(".*\\.rstheme$", boost::regex::icase);
```

Comment:
Idiomatically we usually construct a FilePath object and use a pattern like this: 

https://github.com/rstudio/rstudio/blob/ad3e90de1707b38e7388948e5762d3ce5fed89c3/src/cpp/desktop/DesktopPosixApplication.cpp#L59